pkg/stanza/operator/input/namedpipe/namedpipe.go

```diff
@@ -0,0 +1,196 @@
+)
+
+const (
+	operatorType      = "namedpipe"
+	DefaultMaxLogSize = 1024 * 1024
+)
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
+}
+
+func NewConfig() *Config {
+	return NewConfigWithID(operatorType)
+}
+
+// NewConfig creates a new stdin input config with default values
+func NewConfigWithID(operatorID string) *Config {
+	return &Config{
+		InputConfig: helper.NewInputConfig(operatorID, operatorType),
+	}
+}
+
+// Config is the configuration of a stdin input operator.
+type Config struct {
+	helper.InputConfig `mapstructure:",squash"`
+	BaseConfig         `mapstructure:",squash"`
+}
+
+type BaseConfig struct {
+	Path        string       `mapstructure:"path"`
+	Permissions uint32       `mapstructure:"mode"`
+	Encoding    string       `mapstructure:"encoding"`
+	SplitConfig split.Config `mapstructure:"multiline,omitempty"`
+	TrimConfig  trim.Config  `mapstructure:",squash"`
+}
+
+// Build will build a namedpipe input operator.
+func (c *Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
+	inputOperator, err := c.InputConfig.Build(logger)
+	if err != nil {
+		return nil, err
+	}
+
+	enc, err := decode.LookupEncoding(c.Encoding)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lookup encoding %q: %w", c.Encoding, err)
+	}
+
+	splitFunc, err := c.SplitConfig.Func(enc, true, DefaultMaxLogSize)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create split function: %w", err)
+	}
+
+	return &Input{
+		InputOperator: inputOperator,
+
+		buffer:      make([]byte, DefaultMaxLogSize),
+		path:        c.Path,
+		permissions: c.Permissions,
+		splitFunc:   splitFunc,
+		trimFunc:    c.TrimConfig.Func(),
+	}, nil
+}
+
+type Input struct {
+	helper.InputOperator
+
+	buffer      []byte
+	path        string
+	permissions uint32
+	splitFunc   bufio.SplitFunc
+	trimFunc    trim.Func
+	cancel      context.CancelFunc
+	pipe        *os.File
+	wg          sync.WaitGroup
+}
+
+func (n *Input) Start(p operator.Persister) error {
+	stat, err := os.Stat(n.path)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to stat named pipe: %w", err)
+	}
+
+	if !os.IsNotExist(err) && stat.Mode()&os.ModeNamedPipe == 0 {
+		return fmt.Errorf("path %s is not a named pipe", n.path)
+	}
+
+	if os.IsNotExist(err) {
+		if err := unix.Mkfifo(n.path, n.permissions); err != nil {
+			return fmt.Errorf("failed to create named pipe: %w", err)
+		}
+	}
+
+	// chmod the named pipe because mkfifo respects the umask which may result
+	// in a named pipe with incorrect permissions.
+	if err := os.Chmod(n.path, os.FileMode(n.permissions)); err != nil {
+		return fmt.Errorf("failed to chmod named pipe: %w", err)
+	}
+
+	watcher, err := NewWatcher(n.path)
+	if err != nil {
+		return fmt.Errorf("failed to create watcher: %w", err)
+	}
+
+	pipe, err := os.OpenFile(n.path, os.O_RDWR, os.ModeNamedPipe)
+	if err != nil {
+		return fmt.Errorf("failed to open named pipe: %w", err)
+	}
```

Comment:
There's a lot of complexity here. If reasonable, please make sure unit tests cover these failure modes.

Reply:
Done - added a few more test cases for these various if switches. I'm wondering if it makes sense to abstract this logic out into its own thing that we can test in isolation, outside the operator creation process. WDYT?